tcms/rpc/api/testrun.py

```diff
@@ -223,35 +223,5 @@ def update(run_id, values):
     if not form.is_valid():
         raise ValueError(form_errors_to_list(form))
 
-    test_run = TestRun.objects.get(pk=run_id)
```

Comment:
The reason for your errors is that you are not updating an existing record but trying to create a new one with missing fields. Regardless of what the form says these fields are still required in the model hence the failure.
See how `TestCase.update` method makes sure to reuse the field values which are already in the database.

Reply:
thanks for the tip 👍